enhancements/console/customize-catalogs.md

```diff
@@ -0,0 +1,210 @@
+last-updated: 2022-09-02
+tracking-link: 
+  - https://issues.redhat.com/browse/ODC-6695
+status: implementable
+---
+
+# Customize Catalogs
+
+## Summary
+
+The OpenShift Console provides a "Developer Catalog" which enables 
+the user to create apps and services based on Operator backed services, helm charts, templates, etc.
+
+Cluster administrators can provide additional catalog types (providers) 
+via console plugins, for example by installing an operator. 
+These catalog types (providers) extend the catalog with additional 
+items and are also available as "sub-catalogs".
+
+Some customers want to hide specific types from 
+the catalog or the sub-catalog to streamline their 
+development process without uninstalling a complete operator or disabling the complete console plugin.
+
+In the developer catalog the sub-catalogs are shown as types. 
+The developer catalog shows items from all sub-catalogs.
+When user selects a sub-catalog, they can see a list of 
+available items from this sub-catalog to install.
+
+This enhancement allows cluster admins to selectively 
+disable a sub-catalog and all links to the specific sub-catalog. 
+The developer catalog will also hide any of the items from this sub-catalog.
+
+![Developer catalog and types list](customize-catalogs.png)
+ 
+## Motivation
+
+The developer catalog can be extended by plugins and 
+also dynamically by installing a console plugin or an operator.
+
+Some customers want to align their development process and 
+have teams that support other development teams with their work.
+For this cluster administrators want to hide developer sub-catalog (types) 
+without disabling a full console plugin, for example by uninstalling an operator.
+By hiding for example "Service Backed operators", "Helm", etc. 
+they can still install them via the CLI, see their resources in the topology, 
+but don't show them in the add from developer catalog flow.
+
+### User Stories
+
+#### Story 1
+
+Customer wants to display only their own customized and validated 
+templates on the dev console but they can't change the default devfiles currently.
+
+As a cluster admin, I want to hide Developer catalog for all users.
+For example for customers who has their own way to 
+install workloads, services, etc. without the developer catalog.
+
+#### Story 2
+
+Customer wants to display only their own customized and validated 
+templates on the dev console but they can't change the default devfiles currently.
+
+As a cluster admin, I want to hide certain sub-catalogs (types) for all users.
+For example for customers who don't want to use Helm charts, 
+or another sub-catalog without disabling Helm support in the whole developer console.
+
+### Goals
+
+- Allow cluster admins to hide the developer catalog or certain sub-catalogs (types).
+- Provide information to admins about sub-catalogs (types) they can hide from users.
+
+### Non-Goals
+
+- Allow cluster admins to add enabled sub-catalogs (types) in customization. 
```

Comment:
Can you explain somewhere (maybe in the alternatives) why we are making this a non-goal for now and whether this decision might change in the future?

Reply:
We had the discussion and decided to add `enabledTypes` as well. I updated the document.